apps/prairielearn/elements/pl-checkbox/pl-checkbox.py

```diff
@@ -215,184 +468,123 @@ def render(element_html: str, data: pl.QuestionData) -> str:
-                        + "and <code>n</code> is the total number of options you select."
-                    )
-                else:
-                    raise ValueError(
-                        f"Unknown value for partial_credit_method: {partial_credit_method}"
-                    )
-            else:
-                gradingtext = (
-                    "You must select"
-                    + insert_text
-                    + " You will receive a score of 100% "
-                    + "if you select all options that are true and no options that are false. "
-                    + "Otherwise, you will receive a score of 0%."
-                )
-
-            info_params.update({"gradingtext": gradingtext})
+            info_params["gradingtext"] = generate_grading_text(
+                insert_text=insert_text,
+                num_display_answers=num_display_answers,
+                partial_credit=partial_credit,
+                partial_credit_method=partial_credit_method,
+            )
 
-        with open("pl-checkbox.mustache", encoding="utf-8") as f:
+        with open(CHECKBOX_MUSTACHE_TEMPLATE_NAME, encoding="utf-8") as f:
             info = chevron.render(f, info_params).strip()
 
         html_params = {
             "question": True,
             "name": name,
             "editable": editable,
+            "uuid": pl.get_uuid(),
             "info": info,
             "answers": answerset,
             "inline": inline,
             "hide_letter_keys": pl.get_boolean_attrib(
                 element, "hide-letter-keys", HIDE_LETTER_KEYS_DEFAULT
             ),
+            "helptext": helptext,
         }
 
-        if not hide_help_text:
-            html_params["helptext"] = helptext
-
         if score is not None:
-            try:
-                score = float(score)
-                if score >= 1:
-                    html_params["correct"] = True
-                elif score > 0:
-                    html_params["partial"] = math.floor(score * 100)
-                else:
-                    html_params["incorrect"] = True
-            except Exception as exc:
-                raise ValueError(f"invalid score: {score}") from exc
+            score_type, score_value = pl.determine_score_params(score)
+            html_params[score_type] = bool(score_value)
 
-        with open("pl-checkbox.mustache", encoding="utf-8") as f:
-            html = chevron.render(f, html_params).strip()
+        with open(CHECKBOX_MUSTACHE_TEMPLATE_NAME, encoding="utf-8") as f:
+            return chevron.render(f, html_params).strip()
 
     elif data["panel"] == "submission":
         parse_error = data["format_errors"].get(name, None)
         if parse_error is None:
             partial_score = data["partial_scores"].get(name, {"score": None})
-            feedback = partial_score.get("feedback", None)
+            feedback = cast(
+                "dict[str, Any] | None", partial_score.get("feedback", None)
```

Comment:
```suggestion
                dict[str, Any] | None, partial_score.get("feedback", None)
```
No need to use a string I think.